mne/event.py

```diff
@@ -257,12 +257,19 @@ def read_events(filename, include=None, exclude=None, mask=0):
             raise ValueError('Unknown number of columns in event text file')
 
         event_list = lines[:, goods]
-        if event_list.shape[0] > 0 and event_list[0, 2] == 0:
+        if (mask is not None and event_list.shape[0] > 0 and
+                event_list[0, 2] == 0):
             event_list = event_list[1:]
+            warn('first row of event file discarded (zero-valued)')
 
     event_list = pick_events(event_list, include, exclude)
-    event_list = _mask_trigs(event_list, mask)
-
+    unmasked_len = event_list.shape[0]
+    if mask is not None:
+        event_list = _mask_trigs(event_list, mask)
+        masked_len = event_list.shape[0]
+        if masked_len < unmasked_len:
+            warn('{0} of {1} events masked'.format(unmasked_len - masked_len,
+                                                   unmasked_len))
```

Comment:
In the use case you describe I was thinking of it more like "FYI" rather
than "alert!"
On May 6, 2016 5:49 AM, "Eric Larson" <notifications@github.com> wrote:

> In mne/event.py
> <https://github.com/mne-tools/mne-python/pull/3197#discussion_r62325130>:
>
> >
> >      event_list = pick_events(event_list, include, exclude)
> > -    event_list = _mask_trigs(event_list, mask)
> > -
> > +    unmasked_len = event_list.shape[0]
> > +    if mask is not None:
> > +        event_list = _mask_trigs(event_list, mask)
> > +        masked_len = event_list.shape[0]
> > +        if masked_len < unmasked_len:
> > +            warn('{0} of {1} events masked'.format(unmasked_len - masked_len,
> > +                                                   unmasked_len))
>
> If a user masks with e.g. 1, they would presumably expect all 1 events to
> be dropped, yet they'll get a warning here. I guess we have to choose
> between warning users when they don't need to be warned (the use case I'm
> talking about) versus warning users when they might not realize they're
> losing events (like in the event=0 case), so I can live with the solution
> implemented here..
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/3197/files/878f894c3bce9139d637f4a7439b8f4596929086#r62325130>
>